Add command line arguments to configure leader election options

RaunakShah opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
3. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**

/kind feature


**What this PR does / why we need it**:
This PR adds command line arguments for leader election options - Lease Duration, Renew Deadline and Retry Period. Currently the defaults are 15, 10 and 5 seconds respectively. Without this change, there's no way for a customer/storage plugin to configure those values. 

This PR makes those parameters configurable. Storage plugins deploying external-provisioner can tune those parameters in their YAMLs accordingly. 

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #556

**Special notes for your reviewer**:

Testing:

Created Provisioner with new arguments and verified that Pod is running:
```
# cat vsphere-csi-driver.yaml
...
        - name: csi-provisioner
          args:
            - "--v=4"
            - "--timeout=300s"
            - "--csi-address=$(ADDRESS)"
            - "--kube-api-qps=100"
            - "--kube-api-burst=100"
            - "--leader-election"
            - "--default-fstype=ext4"
            - "leader-election-lease-duration=30"
            - "leader-election-renew-deadline=20"
            - "leader-election-retry-period=10"
...
# kubectl get pod -n vmware-system-csi
NAME                                      READY   STATUS    RESTARTS   AGE
vsphere-csi-controller-74f57f8754-x2bcj   6/6     Running   0          21m
```



**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Add command line arguments to configure leader election options
```

ggriffiths commented:
Looks good once the go fmt issue is fixed

RaunakShah commented:
@ggriffiths yup thats the plan! Thanks for the quick review!

ggriffiths commented:
/assign @xing-yang

xing-yang commented:
Should there be "--" in front of these new options?

```# cat vsphere-csi-driver.yaml
...
        - name: csi-provisioner
          args:
            - "--v=4"
            - "--timeout=300s"
            - "--csi-address=$(ADDRESS)"
            - "--kube-api-qps=100"
            - "--kube-api-burst=100"
            - "--leader-election"
            - "--default-fstype=ext4"
            - "leader-election-lease-duration=30"
            - "leader-election-renew-deadline=20"
            - "leader-election-retry-period=10"
...
```

xing-yang commented:
> there's no way for a customer/storage plugin to configure those values.

My understanding is that these are tunable options when starting the external-provisioner so whoever deploys the CSI driver can configure them?

RaunakShah commented:
@xing-yang 

> Should there be "--" in front of these new options?

Fixed the description! And redeployed with the `--`. 

> My understanding is that these are tunable options when starting the external-provisioner so whoever deploys the CSI driver can configure them?

That's correct, they can be set before creating the pod. I've also added the parameters to the documentation page.

xing-yang commented:
@RaunakShah, I see the following in PR description:
```
This PR adds command line arguments for leader election options - Lease Duration, Renew Deadline and Retry Period. Currently the defaults are 15, 10 and 5 seconds respectively and there's no way for a customer/storage plugin to configure those values.
```
If these options are tunable, why you said "there's no way for a customer/storage plugin to configure those values"?

RaunakShah commented:
>  Do you mean, without this change, they can't be configured; but with this change, they are tunable? 

@xing-yang This is correct, apologies for the confusion. I've updated the PR description with more clarity!

xing-yang commented:
/lgtm
/approve

jingxu97 commented:
@RaunakShah Thanks for the PR. Do you mind making the similar changes to other csi sidecars? Thanks!
